src/main/java/io/kabanero/api/InstanceEndpoints.java

```diff
@@ -233,9 +233,18 @@ public Response setDigestPolicy(@PathParam("instanceName") String instanceName,
         if (instance == null) {
             return Response.status(404).entity(new ResponseMessage(instanceName + " not found")).build();
         }
-
         instance.getSpec().getGovernancePolicy().setStackPolicy(String.valueOf(newPolicy.getPolicy()));
         KabaneroClient.updateInstance(instance);
         return Response.accepted().build();
     }
+
+    @PUT
+    @Path("/{instanceName}/update")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateInstance(@PathParam("instanceName") String instanceName, Kabanero newInstance )
+            throws IOException, ApiException, GeneralSecurityException {
+        KabaneroClient.updateInstance(newInstance);
```

Comment:
lets add some checking before updating maybe something simple to start like:

1. Checking newInstance isn't null
1. checking the instance name path param to the newInstance name to ensure they match before moving along

Reply:
also we're going to need to put this endpoint behind the protected endpoints class. We should def make sure a user is "admin" before allowing this type of update.

There should be code already in the protected endpoints folder that checks for isAdmin() so you can follow that path. I would suggest creating another file `src/main/java/io/kabanero/api/restricted/restrictedInstanceEndpoints.java` to put this 1 endpoint in